core/arch/aarch64/instr_create.h

```diff
@@ -369,8 +369,11 @@
  * they just need to know whether they need to preserve the app's flags, so maybe
  * we can just document that this may not write them.
  */
-#define XINST_CREATE_slr_s(dc, d, s) \
-  instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (s))
+#define XINST_CREATE_slr_s(dc, d, rm_or_imm) \
+  opnd_is_reg(rm_or_imm) ? \
+    instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (rm_or_imm)) : \
+    instr_create_1dst_3src((dc), OP_ubfm, (d), (d), (rm_or_imm), \
```

Comment:
I disagree with the parentheses comments here: even where not technically needed like ", (d), ", it is all too easy to later edit the macro and add something in between the commas and forget to add the parens, or to copy-paste a param name somewhere else where parens are needed, so IMHO it is not a waste to have them there.

Re: the line length exclusion: feel free to remove aarch64/instr_create.h from the list.  I planned to have some files like the x86 and arm decode tables and probably those arch's instr_create.h permanently excluded.